webserver/src/macros_local.rs

```diff
@@ -36,3 +36,19 @@ macro_rules! run_async_query {
         .expect("internal server error")
     }};
 }
+
+#[macro_export]
+macro_rules! run_async_typesense_query {
+    ($search_state:expr, $query:expr, $url:expr, $user:expr) => {{
+        use reqwest::Error;
+        use actix_web::error::ErrorInternalServerError;
+
+        web::block(move || {
+            let response = $query(&$search_state,  $url, $user).map_err(Error::from);
+            response
+        })
+        .await
+        .map_err(ErrorInternalServerError)
+        .expect("internal server error")
```

Comment:
Don't use ".expect()" use "?"

And we probably don't need this macro, as run_async_query does the same thing

Reply:
> Don't use ".expect()" use "?"

This has been addressed in the new pull request.
I believe we need the new marco for handling different return error types.